### PR TITLE
Support using table functions with multiple columns in select list

### PR DESCRIPTION
**Pull request by mfussenegger**
## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: sql/src/main/java/io/crate/protocols/postgres/types/PGTypes.java [hunk 1]
```diff
@@ -97,31 +102,46 @@
...
-    public static PGType get(DataType type) {
-        if (type.id() == ArrayType.ID) {
-            DataType<?> innerType = ((ArrayType) type).innerType();
-            if (innerType.id() == ArrayType.ID) {
-                // if this is a nested collection stream it as JSON because
-                // postgres binary format doesn't support multidimensional arrays
-                // with sub-arrays of different length
-                // (something like [ [1, 2], [3] ] is not supported)
+    public static PGType get(DataType<?> type) {
+        switch (type.id()) {
```
Comment (kovrus) [inline sql/src/main/java/io/crate/protocols/postgres/types/PGTypes.java]: Do we need to handle the array of row type separately, such as we have a special handling for [RowType](https://github.com/crate/crate/pull/9627/files#diff-4af35b46ab2acffaf1a354f41e708866R133).

File: common/src/main/java/io/crate/types/RowType.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+    public int numElements() {
+        return fieldTypes.size();
+    }
+
+    public DataType<?> getFieldType(int position) {
+        return fieldTypes.get(position);
+    }
+
+    public String getFieldName(int position) {
```
Comment (kovrus) [inline common/src/main/java/io/crate/types/RowType.java]: should we have some pre condition here to be more safe and get a better exception message if we get out of bound exception.

File: common/src/main/java/io/crate/types/RowType.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+    @Override
+    public Row value(Object value) throws IllegalArgumentException, ClassCastException {
+        return (Row) value;
+    }
+
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    @Override
+    public int compareValueTo(Row val1, Row val2) {
+        for (int i = 0; i < fieldTypes.size(); i++) {
```
Comment (kovrus) [inline common/src/main/java/io/crate/types/RowType.java]: not sure, maybe we can first`Row#numColumns`  of val1 and val2?

File: common/src/main/java/io/crate/types/DataTypes.java [hunk 4]
```diff
@@ -314,6 +315,7 @@ private static boolean safeConversionPossible(DataType type1, DataType type2) {
         entry("int4", INTEGER),
         entry("int8", LONG),
         entry("name", STRING),
+        entry("regproc", STRING),
```
Comment (kovrus) [inline common/src/main/java/io/crate/types/DataTypes.java]: shall we add `regproc` to aliases in data-types.rst?

File: common/src/main/java/io/crate/types/RowType.java (hunk 2 above)
Comment (mfussenegger) [reply common/src/main/java/io/crate/types/RowType.java]: Not sure, I think the out-of-bounds exception that will be raised is fairly fitting.

File: common/src/main/java/io/crate/types/RowType.java (hunk 3 above)
Comment (mfussenegger) [reply common/src/main/java/io/crate/types/RowType.java]: I added an assertion to make sure they have the same size. I think comparing different rows to each other shouldn't be allowed. (This here is for sorting, and we can't sort different row values).

File: sql/src/main/java/io/crate/protocols/postgres/types/PGTypes.java (hunk 1 above)
Comment (mfussenegger) [reply sql/src/main/java/io/crate/protocols/postgres/types/PGTypes.java]: Good point. Adding a fixup

**Pull request merged**